apps/src/dance/Dance.js

```diff
@@ -384,7 +384,13 @@ Dance.prototype.onP5Preload = function () {
   Sounds.getSingleton().register(options);
   const getSelectedSong = () => getStore().getState().selectedSong;
 
-  this.nativeAPI = new DanceParty(this.p5, getSelectedSong, audioCommands.playSound, this.onPuzzleComplete.bind(this));
+  this.nativeAPI = new DanceParty(
+    this.p5,
+    getSelectedSong,
+    audioCommands.playSound,
```

Comment:
For https://github.com/code-dot-org/dance-party/pull/23#discussion_r227087754, I wonder if we should do something like:

```js
const boundPlaySong  = (...args) => {
  audioCommands.playSound.apply(null, {
    onEnded: () => this.studioApp_.toggleRunReset('run'),
    ...args,
  });
}
```

And pass in `boundPlaySong` instead of `audioCommands.playSound`.  Just a first pass, let me know if this seems like the right direction or not!